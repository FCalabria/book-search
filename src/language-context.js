import React from 'react';

export const languages = {
  es: {
    search: 'Busca un libro',
    similar: 'Buscar parecido',
    searching: 'Buscando',
    searchError: 'No hemos encontrado nada con'
  },
  en: {
    search: 'Search a book',
    similar: 'Search similar',
    searching: 'Searching',
    searchError: 'We could not find anything by'
  },
};

const LanguageContext = React.createContext(languages.es);
export const LanguageProvider = LanguageContext.Provider;
export const LanguageConsumer = LanguageContext.Consumer;