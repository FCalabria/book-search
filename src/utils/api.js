async function apiFetch (path, fetchOptions) {
  try {
    const result = await fetch(`http://openlibrary.org/${path}`, fetchOptions)
    if (result.ok) {
      const response = await result.json()
      return Promise.resolve(response)
    } else {
      return Promise.reject(result)
    }
  } catch (error) {
    console.log(`Fetch error ${path}`, error)
    return Promise.reject(new Error('Network error: check your connection and try again'))
  }
}

export default apiFetch