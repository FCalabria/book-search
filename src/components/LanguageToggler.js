import React from 'react';
import {LanguageConsumer} from '../language-context';

function LanguageToggler() {
  const renderOptions = values => values.map(value => (
    <option value={value} key={value}>{value}</option>
  ))
  return (
    <LanguageConsumer>
      {({keys, selected, toggleLanguage}) => (<select
        className="absolute top-0 right-0 mt-3 mr-3 text-gray-700 bg-teal-100 border-pink-200 border-2 shadow-lg"
        defaultValue={selected}
        onChange={e => toggleLanguage(e.target.value)}
      >
        {renderOptions(keys)}
        </select>
      )}
    </LanguageConsumer>
    );
}

export default LanguageToggler;

