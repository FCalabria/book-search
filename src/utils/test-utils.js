import React from 'react'
import { render } from '@testing-library/react'
import {LanguageProvider, languages} from '../language-context'


const AllTheProviders = ({ children }) => {
  return (
    <LanguageProvider value={{
      language: languages.ES,
      keys: Object.keys(languages),
      selected: 'ES',
      toggleLanguage: () => {}
    }}>
      {children}
    </LanguageProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
