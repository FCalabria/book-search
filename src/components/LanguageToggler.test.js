import React from 'react';
import { render } from '../utils/test-utils';
import LanguageToggler from './LanguageToggler';
import {languages} from '../language-context'


describe.only('LanguageToggler', () => {
  test('should render correctly', () => {
    const container = render(<LanguageToggler />).container
    expect(container.firstChild).toMatchSnapshot();
  });
})
