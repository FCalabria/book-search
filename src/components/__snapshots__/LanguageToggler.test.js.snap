// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageToggler should render correctly 1`] = `
<select
  class="absolute top-0 right-0 mt-3 mr-3 text-gray-700 bg-teal-100 border-pink-200 border-2 shadow-lg"
>
  <option
    selected=""
    value="ES"
  >
    ES
  </option>
  <option
    value="EN"
  >
    EN
  </option>
</select>
`;
